packages/lesswrong/components/common/CloudinaryImage2.tsx

```diff
@@ -20,46 +23,71 @@ function cloudinaryPropsToStr(props) {
   return sb.join(",");
 }
 
+function makeCloudinaryImageUrl (publicId: string, cloudinaryProps: CloudinaryPropsType) {
+  return `https://res.cloudinary.com/${cloudinaryCloudNameSetting.get()}/image/upload/${cloudinaryPropsToStr(cloudinaryProps)}/${publicId}`
+}
+
 // Cloudinary image without using cloudinary-react. Allows SSR. See:
 // https://github.com/LessWrong2/Lesswrong2/pull/937 "Drop cloudinary react"
 // https://github.com/LessWrong2/Lesswrong2/pull/964 "Temporarily revert removal of cloudinary-react"
-const CloudinaryImage2 = ({width, height, objectFit, publicId, imgProps, className}: {
+const CloudinaryImage2 = ({width, height, objectFit, publicId, imgProps, header, className}: {
   width?: number|string,
   height?: number,
-  objectFit?: string,
+  objectFit?: 'fill' | 'contain' | 'cover' | 'none' | 'scale-down',
   publicId: string,
   imgProps?: CloudinaryPropsType,
+  header?: boolean,
   className?: string,
 }) => {
-  const cloudinaryCloudName = cloudinaryCloudNameSetting.get()
-
-  let cloudinaryProps: any = {
+  let cloudinaryProps: CloudinaryPropsType = {
     c: "fill",
     dpr: "auto",
     g: "custom",
     q: "auto",
     f: "auto"
   };
-  let imageStyle: any = {};
+  let imageStyle: CSSProperties = {};
 
   if (width) {
-    cloudinaryProps.w = width
+    cloudinaryProps.w = width.toString()
     imageStyle.width = width
   }
   if (height) {
-    cloudinaryProps.h = height;
+    cloudinaryProps.h = height.toString()
     imageStyle.height = height+"px";
   }
+  // ignore input width if we're told we have a header
+  if (header) {
+    cloudinaryProps.h = ((height || DEFAULT_HEADER_HEIGHT)*2).toString()
```

Comment:
This line and the next one are basically never going to be used, right? Because every browser supports srcSet, so imageUrl is not going to be read?

If that's the case I would vote for just deleting them – I feel like they cause more confusion than help.

Reply:
If your monitor is wider than 3000px, it'll still use this.

Reply:
I tried removing it, but felt like it was easier on the devs mind to tell them that src is being set to something sensible but ignored, rather than to just have src be unmodified and then have the dev tempted to try to double check that in fact src was never used except by IE. ¯\\\_(ツ)_/¯